src/WebJobs.Extensions.DurableTask/Listener/TaskOrchestrationShim.cs

```diff
@@ -178,6 +179,18 @@ public override async Task<string> Execute(OrchestrationContext innerContext, st
                     this.context.IsReplaying));
             }
 
+#if !FUNCTIONS_V1
+            var activity = Activity.Current;
+            /* activity may be null when running this repo's tests but does not appear
+             * empty when actually running the samples
+             */
+            if (activity != null)
+            {
+                activity.AddTag("DurableFunctionsType", "Orchestration");
+                activity.AddTag("DurableOrchestrationInstanceId", this.context.InstanceId);
+                activity.AddTag("DurableFunctionState", this.context.GetSerializedCustomStatus());

```

Comment:
I think the "state" should be one of [these](https://github.com/Azure/azure-functions-durable-extension/blob/dev/src/WebJobs.Extensions.DurableTask/OrchestrationRuntimeStatus.cs).

The custom status could be useful, but it can get quite large. I think we can add it later based on feedback.

Reply:
Good point, I incorporated these changes, to the extend possible, more details coming up in a separate comment